cluster.go

```diff
@@ -281,6 +292,41 @@ func (c *Cluster) Sync() error {
 	return err
 }
 
+func (c *Cluster) traceTopoChanged(prevTopo ClusterTopo, newTopo ClusterTopo) {
+	if c.co.ct.TopoChanged != nil {
+		var addedNodes []trace.ClusterNodeInfo
+		var removedNodes []trace.ClusterNodeInfo
+
+		prevTopoMap := prevTopo.Map()
+
+		for addr, newNode := range newTopo.Map() {
+			if prevNode, ok := prevTopoMap[addr]; ok {
+				if !reflect.DeepEqual(newNode, prevNode) {
+					// Previous Node changed its information
+					removedNodes = append(removedNodes, trace.ClusterNodeInfo{
+						Addr:      prevNode.Addr,
+						Slots:     prevNode.Slots,
+						IsPrimary: prevNode.SecondaryOfAddr == "",
+					})
+					addedNodes = append(addedNodes, trace.ClusterNodeInfo{
+						Addr:      newNode.Addr,
+						Slots:     newNode.Slots,
+						IsPrimary: newNode.SecondaryOfAddr == "",
+					})
+				}
+			} else {
+				addedNodes = append(addedNodes, trace.ClusterNodeInfo{
+					Addr:      newNode.Addr,
+					Slots:     newNode.Slots,
+					IsPrimary: newNode.SecondaryOfAddr == "",
+				})
+			}
+		}
+
+		c.co.ct.TopoChanged(trace.ClusterTopoChanged{Added: addedNodes, Removed: removedNodes})
```

Comment:
You forgot to check for removed nodes. Right now you're only tracking added and changed nodes.

Reply:
Oh, my mistake! I'll check the added, removed, changed nodes.